utils/allowCors.js

```diff
@@ -0,0 +1,25 @@
+export default function allowCors(fn, allow = false) {
+  const whiteListedDomains = ["app.gmx.io", "gmx-interface.pages.dev"]
+  return async (req, res) => {
+    res.setHeader("Access-Control-Allow-Credentials", true)
+    if (allow) {
+      res.setHeader("Access-Control-Allow-Origin", "*")
+    } else if (whiteListedDomains.some(d => req.headers.origin.includes(d))) {
```

Comment:
is the `else if (whitelistedDomains.some ...)` condition needed? would just doing `else { ...` work?

Reply:
We need the white-listed domain to be considered when adding the headers so I feel we will need the else if.